rust/main/agents/relayer/src/relayer.rs

```diff
@@ -568,14 +582,27 @@ impl BaseAgent for Relayer {
             .iter()
             .map(|(key, origin)| (key.id(), origin.prover_sync.clone()))
             .collect();
-        let relayer_router = relayer_server::Server::new(self.destinations.len())
+
+        let mut server_builder = relayer_server::Server::new(self.destinations.len())
             .with_op_retry(sender.clone())
             .with_message_queue(prep_queues)
             .with_dbs(dbs)
             .with_gas_enforcers(gas_enforcers)
             .with_msg_ctxs(msg_ctxs)
-            .with_prover_sync(prover_syncs)
-            .router();
+            .with_prover_sync(prover_syncs);
+
+        // Set kaspa_db to server_builder from dymension_args provider if available
+        if let Some(dym_args) = &self.dymension_kaspa_args {
+            if let Some(kaspa_db) = dym_args.kas_provider.kaspa_db() {
+                server_builder = server_builder.with_kaspa_db(kaspa_db.clone());
+            } else {
+                error!("Kaspa DB missing in Kaspa provider when setting up relayer server");
+            }
+        } else {
+            error!("Dymension Kaspa args missing when setting up relayer server");
+        }
```

Comment:
if with_kaspa_db took an option argument the code would be much shorter and cleaner